R/checkers.R

```diff
@@ -45,3 +56,345 @@ check_elements_non_neg <- function(x, arg = "x", call = rlang::caller_env()) {
   }
   invisible()
 }
+
+#' Check that the argument is of type date
+#'
+#' @param x Object with type checking, here this will be a date
+#' @param arg Name of the argument supplying the object
+#' @param call Calling environment to be passed to [cli::cli_abort()]
+#'
+#' @return NULL, invisibly
+check_date <- function(x, arg = "x", call = rlang::caller_env()) {
+  if ((!rlang::is_integerish(x)) || (!inherits(x, "Date"))) {
+    throw_type_error(
+      object = x,
+      arg_name = arg,
+      expected_type = "Date",
+      call = call
+    )
+  }
+  invisible()
+}
```

Comment:
I'm surprised we need to create all these type checkers for ourselves. Is there not a standard tidyverse or tidyverse-adjacent library for this? 

cc @seabbs

Reply:
You may want [checkmate ](https://mllg.github.io/checkmate/)

Reply:
Definitely! I didn't realize this existed, it looks very useful!

Reply:
Ok my only complaint/thought here is by using these, I don't see an argument to pass a more helpful errror message (e.g. based on the `arg`). So as written, it will tell you which argument failed the test, but I think if I remove the `check_{date/vectors/nonmissinginess}` I lose this functionality and will just get an error about that type but not specific to the `arg`

Reply:
The `assert_` functions from `checkmate` give you sume but not total opportunity to customize error messages. if you want more control over the message you can use `check_*()` rather than `assert_*()` and then raise the errors yourself.